mpire/comms.py

```diff
@@ -60,49 +61,48 @@ def __init__(self, ctx: mp.context.BaseContext, n_jobs: int, order_tasks: bool)
         self._initialized = False
 
         # Whether or not to inform the child processes to keep order in mind (for the map functions)
-        self._keep_order = self.ctx.Event()
+        self._keep_order = self.ctx.Value(ctypes.c_bool, False, lock=True)
 
         # Queue to pass on tasks to child processes. We keep track of which worker completed the last task and which
         # worker is working on what task
         self._task_queues: List[mp.JoinableQueue] = []
-        self._task_idx: int = None  # type: ignore
+        self._task_idx: Optional[int] = None
         self._worker_running_task_locks: List[mp.RLock] = []
         self._worker_running_task: List[mp.Value] = []
         self._last_completed_task_worker_id = collections.deque()
-        self._worker_working_on_job: List[mp.Value] = []
+        self._worker_working_on_job: Optional[mp.Array] = None
```

Comment:
If you use `from __future__ import annotations` I think you can already use `mpArray | None` instead of `Optional[mp.Array]` even in py3.8

Reply:
Let's use that for another PR ;) 

I'm also still planning to introduce pre-commit (with black formatting, etc), pytest, and let's definitely add that new style of annotations to it. 

Will do so PR by PR